src/lib/route.js

```diff
@@ -56,11 +66,20 @@ class Route {
   combine (alternateRoute) {
     const combinedCurve = this.curve.combine(alternateRoute.curve)
     const combinedHops = this._simpleHops()
+    const havePath = {}
+
+    for (let list of [this.paths, alternateRoute.paths]) {
+      for (let path of list) {
+        havePath[path] = true
+      }
+    }
+
     return new Route(combinedCurve, combinedHops, {
       minMessageWindow: Math.max(this.minMessageWindow, alternateRoute.minMessageWindow),
       isLocal: false,
 
-      addedDuringEpoch: Math.max(alternateRoute.addedDuringEpoch, this.addedDuringEpoch)
+      addedDuringEpoch: Math.max(alternateRoute.addedDuringEpoch, this.addedDuringEpoch),
+      paths: Object.keys(havePath)
```

Comment:
This is going to be a 1 dimensional array, because Object.keys always returns an array of strings